### PR TITLE
Get timestamp of last completed poll of the blockchain

### PR DESCRIPTION
**Pull request by jp1ac4**
This is a first step towards #1373.

The timestamp of the last completed poll of the blockchain will be stored in the database, and this value will be made available via the `getinfo` command.

### CODE REVIEW

**Comment by jp1ac4** [comment]
I added a commit to ensure the poller doesn't run while the blockchain is syncing. This was already the case for our regular polling, but if a user were to broadcast a spend while syncing is in progress, which could be done only via liana-cli currently, the poller would indeed run.

Running the poller during syncing could take a while to finish as the tip height would be more likely to change before completion, causing the poller to restart each time. We may want to allow this in future (e.g. to sync the wallet during IBD?) but for now I think the poller behaviour should be consistent between regular polling and that triggered by a spend broadcast.

This change also ensures that we only store the last poll timestamp for a poll against a synced blockchain, which is the implicit assumption for downstream logic in the GUI, i.e. when the first poll completes, the wallet is synced. Although we don't want downstream logic to determine the upstream implementation, I don't think storing the timestamp of a poll against a non-synced blockchain would be very useful on its own.

**Comment by jp1ac4** [comment]
I made a couple of changes to the sqlite method `set_wallet_last_poll_timestamp()`:
- update a single row in wallets table where `id = WALLET_ID`
- return `Result<(), SqliteDbError>`

File: src/commands/mod.rs [hunk 1]
```diff
@@ -324,6 +324,7 @@ impl DaemonControl {
             },
             rescan_progress,
             timestamp: db_conn.timestamp(),
+            last_poll_timestamp: db_conn.last_poll_timestamp(),
```
Comment (edouardparis) [inline src/commands/mod.rs:327]: seems to me that we do two database call to get db_wallet for timestamp() and last_poll_timestamp(), can we have instead wallet = db_conn.wallet() and then wallet.timestamp and wallet.last_poll_timestamp

Comment (jp1ac4) [reply src/commands/mod.rs:327]: Yes I think that would be better. I'll add a new `wallet()` method to the DB interface.

**Comment by jp1ac4** [comment]
I've added a `Wallet` struct and corresponding `wallet()` method to the database interface.

Perhaps I should use `u32` for `last_poll_timestamp` for consistency with the other timestamps. I did consider it previously, but now we have the `Wallet` struct the inconsistency is more apparent.

**Comment by edouardparis** [comment]
yes, I agree for consistency u32 would be better

**Comment by jp1ac4** [comment]
I changed the `last_poll_timestamp` to `u32`.

**Review by edouardparis — APPROVED** [review]
ACK dee9554c518793b57b71965d835905300ef86fbc

**Pull request merged**